Add forward_prefetch flag to fully_shard

rohan-varma opened the pull request:
Summary: Per title

Test Plan: CI

Differential Revision: D44656110

awgu reviewed (approved):
I am stamping to unblock, but the lint error is real.

rohan-varma committed:
Add forward_prefetch flag to fully_shard (#98277)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/98277

Per title

Test Plan: CI

Differential Revision: D44656110

fbshipit-source-id: 417165bd8e99a4311a223430cec93f98a0e0f4e4

rohan-varma commented:
@pytorchbot merge -f "CI done"
